dataplicity/device_meta.py

```diff
@@ -36,3 +38,15 @@ def get_os_version():
     # Linux is a fair assumption for now
     distro = " ".join(platform.linux_distribution()).strip()
     return distro
+
+def get_ip_address_list():
+    # Get the ip addresses from all the interfaces
+    try:
+        interfaces = get_all_interfaces()
+    except Exception:
```

Comment:
This should be a bear accept to catch all exceptions.

Reply:
I think we started with that approach, @meshy suggested making it a bit specific.  I'm not sure the difference - are there any Pythonic exceptions which _don't_ device from Exception?

Reply:
Yes. https://docs.python.org/2/library/exceptions.html#exception-hierarchy

Not likely to occur. But if you are going to catch em all, you want to catch em all.

Reply:
And we should log this...

```python
log.exception("Error getting interfaces")
```

Reply:
huh, didn't know that.  Nonetheless looking at those, I'm not sure we want to catch those....

BaseException
 +-- SystemExit
 +-- KeyboardInterrupt
 +-- GeneratorExit

Those sorts of errors should probably bubble up and be handled at the top of the code no?  The only ones we want to squash relate directly to the calling the C function here.  I _think_.

Reply:
Agree re: logs.

Reply:
Maybe. Although I think that method is wrapped higher up in a bear accept. If its in a thread it shouldn't be an issue, since SystemExits get raised in the main thread.

If I could re-write that function, we would be more confident in what exceptions if any it can raise.

Reply:
Although the original call to get_meta is wrapped in a bare except, I wrapped this separately so that if this fails, the other meta data will still be sent.